src/query/atomWithQuery.ts

```diff
@@ -88,7 +89,7 @@ export function atomWithQuery<
   createQuery: CreateQueryOptions<
     AtomWithQueryOptions<TQueryFnData, TError, TData, TQueryData, TQueryKey>
   >,
-  getQueryClient: GetQueryClient = (get) => get(queryClientAtom)
+  getQueryClient: GetQueryClient = defaultGetQueryClient
```

Comment:
We don't want this change, if this is required, then it probably means `<Provider>` doesn't work and it's breaking change.